### PR TITLE
Reducing data race via SyncLockMap

### PR DESCRIPTION
**Pull request by Mzack9999**
## Proposed changes
This PR replace `contextargs.Args` with `mapsutil.SyncLockMap` improving native race avoidance

## Checklist
- [ ] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [ ] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: v2/pkg/protocols/common/contextargs/contextargs.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ type Context struct {
 	// Access to Args must use lock strategies to prevent data races
 	*sync.RWMutex
 	// Args is a workflow shared key-value store
```
Comment (tarunKoyalwar) [inline v2/pkg/protocols/common/contextargs/contextargs.go:20]: unused mutex , `SyncLockMap` internally already has a mutex

File: v2/pkg/protocols/common/contextargs/args.go [hunk 2]
```diff
@@ -26,6 +28,6 @@ func (Args Args) IsEmpty() bool {
 }
 
 // create a new args map instance
-func newArgs() map[string]interface{} {
-	return make(map[string]interface{})
+func newArgs() *mapsutil.SyncLockMap[string, interface{}] {
+	return &mapsutil.SyncLockMap[string, interface{}]{Map: mapsutil.Map[string, interface{}]{}}
 }
```
Comment (tarunKoyalwar) [inline v2/pkg/protocols/common/contextargs/args.go:33]: except `newargs()` all types,methods in this file not used anymore . we can move it contextargs.go and delete this file

**Review by tarunKoyalwar — CHANGES_REQUESTED** [review]
suggesting removing unused types,methods

File: v2/pkg/protocols/common/contextargs/args.go (hunk 2 above)
Comment (Mzack9999) [reply v2/pkg/protocols/common/contextargs/args.go]: nice one, forgot to cleanup!

**Review by tarunKoyalwar — APPROVED** [review]

**Pull request merged**